tornado-drivers/ptx-jni/src/main/cpp/source/PTXModule.cpp

```diff
@@ -75,6 +91,102 @@ JNIEXPORT jbyteArray JNICALL Java_uk_ac_manchester_tornado_drivers_ptx_PTXModule
     return from_module(env, &module);
 }
 
+/*
+ * Class:     uk_ac_manchester_tornado_drivers_ptx_PTXModule
+ * Method:    cuModuleLoadDataEx
+ * Signature: ([BLjava/lang/String;)[B
+ */
+JNIEXPORT jbyteArray JNICALL Java_uk_ac_manchester_tornado_drivers_ptx_PTXModule_cuModuleLoadDataEx
+  (JNIEnv *env, jclass clazz, jbyteArray source, jstring compilerFlags) {
+    CUresult result;
+
+    size_t ptx_length = env->GetArrayLength(source);
+#ifdef _WIN32
+    char *ptx = new char[ptx_length + 1];
+#else
+    char ptx[ptx_length + 1];
+#endif
+    env->GetByteArrayRegion(source, 0, ptx_length, reinterpret_cast<jbyte *>(ptx));
+    ptx[ptx_length] = 0; // Make sure string terminates with a 0
+
+    const char* chars = env->GetStringUTFChars(compilerFlags, nullptr);
+    std::string flags(chars);
+    env->ReleaseStringUTFChars(compilerFlags, chars);
+
+    std::istringstream iss(flags);
+    std::vector<CUjit_option> options;
+    std::vector<unsigned int> values;
+
+    std::string token;
+    std::vector<std::pair<std::string, std::string>> entries;
+    while (iss >> token) {
+        if (token.rfind("CU_JIT", 0) == 0) {
+            std::string val;
+            if (!(iss >> val)) {
+                throw std::runtime_error("Missing value for flag: " + token);
+            }
+            entries.emplace_back(token, val);
+        } else {
+            std::cerr << "Skipping unsupported CUDA JIT flag: " << token << "\n";
+        }
+    }
+    options.reserve(entries.size());
+    values.reserve(entries.size());
+
+    for (const auto &p : entries) {
+        const std::string &flagName = p.first;
+        const std::string &valStr = p.second;
+
+        auto it = CUDAJITFlagsMap.find(flagName);
+        if (it == CUDAJITFlagsMap.end()) {
+            std::cerr << "Unsupported CUDA JIT flag: " << flagName << "\n";
+            continue;
+        }
```

Comment:
why do we have both a map in C++ and one in Java? 

See in `PTXCodeCache.java`:
```java
if (!SUPPORTED_PTX_JIT_FLAGS.contains(flag)) {
    throw new TornadoBailoutRuntimeException("Unsupported PTX JIT compiler flag: " + flag + ". Supported flags are: " + SUPPORTED_PTX_JIT_FLAGS);
}
```

It seems redundant.